envoy/Makefile

```diff
@@ -79,7 +79,7 @@ all: clean-bins release
 else
 BAZEL_BUILD_OPTS =
 
-all: clean-bins envoy $(GO_TARGETS)
+all: envoy $(GO_TARGETS)
```

Comment:
Need to remove the binaries to force a re-link, otherwise the version check will fail.

Reply:
Ah. In my setup, this saves an extra few seconds. Probably if I really want to get those back I'd need to look at introducing an option to avoid the version check. I don't plan on doing that at this stage.

Reply:
@jrajahalme I wonder if something like this could work for local builds?

```
 $ git di
diff --git a/envoy/Makefile b/envoy/Makefile
index 2be0db3a5fe5..f0d94b813ece 100644
--- a/envoy/Makefile
+++ b/envoy/Makefile
@@ -80,7 +80,7 @@ else
 BAZEL_OPTS ?=
 BAZEL_BUILD_OPTS =
 
-all: clean-bins envoy $(GO_TARGETS)
+all: maybe-clean-bins envoy $(GO_TARGETS)
 endif
 
 debug: envoy-debug $(GO_TARGETS)
@@ -132,6 +132,9 @@ bazel-restore: $(BAZEL_ARCHIVE)
        -mkdir $(dir $(BAZEL_CACHE))
        cd $(dir $(BAZEL_CACHE)) && sudo tar xjf $(BAZEL_ARCHIVE) --warning=no-timestamp
 
+maybe-clean-bins: ../.git
+       $(MAKE) clean-bins
+
 # Remove the binaries to get fresh version SHA
 clean-bins: force
        -rm -f $(ENVOY_BINS)
```